--- gc-7.3alpha2/configure.ac.bero	2013-01-15 21:30:39.968204258 +0100
+++ gc-7.3alpha2/configure.ac	2013-01-15 21:30:49.477089777 +0100
@@ -21,7 +21,7 @@ AC_PREREQ(2.61)
 AC_REVISION($Revision: 1.69 $)
 GC_SET_VERSION
 AM_INIT_AUTOMAKE([foreign dist-bzip2 nostdinc])
-AM_CONFIG_HEADER([include/private/config.h])
+AC_CONFIG_HEADERS([include/private/config.h])
 AM_MAINTAINER_MODE
 
 AC_SUBST(PACKAGE)
